charmcraft/linters.py

```diff
@@ -0,0 +1,64 @@
+# Copyright 2021 Canonical Ltd.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# For further info, check https://github.com/canonical/charmcraft
+
+"""Analyze and lint charm structures and files."""
+
+import os
+import shlex
+from collections import namedtuple
+
+CheckType = namedtuple("CheckType", "trait warning error")(
+    trait="trait", warning="warning", error="error"
+)
+
+
+class Language:
+    """Check the language used to write the charm.
+
+    Currently only Python is detected, if the following checks are true:
+
+    - the charm has a text dispatch with a python call
+    - the charm has a `.py` entry point
+    - the entry point file is executable
+    """
+
+    check_type = CheckType.trait
+    name = "language"
+    url = "https://juju.is/docs/sdk/charmcraft-analyze#heading--language"  # FIXME: confirm url
+    text = "The charm is written with Python."
+
+    # different result constatns
+    PYTHON = "python"
```

Comment:
enums in python are pretty good these days, maybe group them with that like on https://github.com/canonical/craft-parts/blob/main/craft_parts/actions.py ?

Reply:
You say to build something like the following?
```
class Language:
    ....
    class Results(enum.Enum):
        PYTHON = "python"
        UNKNOWN = "unknown"
    ...
```
Then, to use it we would need to do:
```
>>> Language.Results.PYTHON.value
'python'
```

It's more verbose, but what I really dislike is having to access the `.value` attribute.

If you really want to group them, I'd use:
```
class Language:
    ....
    Result = namedtuple("Result", "python unknown")(python="python", unknown="unkwnon")
    ...
```
Then, it's just:
```
>>> Language.Result.python
'python'
```

What do you think?

Reply:
namedtuple gets the job done too and is used across this code base so yeah, +1

Reply:
Improved.